drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-backend/src/main/java/org/drools/workbench/screens/scenariosimulation/backend/server/ScenarioValidationService.java

```diff
@@ -0,0 +1,244 @@
+import static org.drools.scenariosimulation.backend.util.DMNSimulationUtils.extractDMNRuntime;
+import static org.drools.scenariosimulation.backend.util.ScenarioBeanUtil.fillBean;
+import static org.drools.scenariosimulation.backend.util.ScenarioBeanUtil.loadClass;
+import static org.drools.scenariosimulation.backend.util.ScenarioBeanUtil.navigateToObject;
+import static org.drools.workbench.screens.scenariosimulation.backend.server.util.DMNTypeUtils.getRootType;
+import static org.drools.workbench.screens.scenariosimulation.model.FactMappingValidationError.createFieldChangedError;
+import static org.drools.workbench.screens.scenariosimulation.model.FactMappingValidationError.createGenericError;
+import static org.drools.workbench.screens.scenariosimulation.model.FactMappingValidationError.createNodeChangedError;
+import static org.kie.dmn.feel.lang.types.BuiltInType.CONTEXT;
+import static org.kie.dmn.feel.lang.types.BuiltInType.UNKNOWN;
+
+@ApplicationScoped
+public class ScenarioValidationService
+        extends AbstractKieContainerService {
+
+    /**
+     * Validate the structure of a simulation. It does not validate the content of the cells
+     * @param simulation to validate
+     * @param path to test scenario file
+     * @return list of validation errors
+     */
+    public List<FactMappingValidationError> validateSimulationStructure(Simulation simulation, Path path) {
+        KieContainer kieContainer = getKieContainer(path);
+        ScenarioSimulationModel.Type type = simulation.getSimulationDescriptor().getType();
+        if (DMN.equals(type)) {
+            return validateDMN(simulation, kieContainer);
+        } else if (RULE.equals(type)) {
+            return validateRULE(simulation, kieContainer);
+        } else {
+            throw new IllegalArgumentException("Only DMN and RULE test scenarios can be validated");
+        }
+    }
+
+    /**
+     * Validate structure of a DMN test scenario.
+     * Supported checks for each column:
+     * - empty column skip
+     * - DMN node removed
+     * - simple type becomes complex type
+     * - navigation of data type still valid
+     * - field type changed
+     * @param simulation
+     * @param kieContainer
+     * @return
+     */
+    protected List<FactMappingValidationError> validateDMN(Simulation simulation, KieContainer kieContainer) {
+        List<FactMappingValidationError> errors = new ArrayList<>();
+        String dmnFilePath = simulation.getSimulationDescriptor().getDmnFilePath();
+        DMNModel dmnModel = getDMNModel(kieContainer, dmnFilePath);
+
+        for (FactMapping factMapping : simulation.getSimulationDescriptor().getFactMappings()) {
+            if (isToSkip(factMapping)) {
+                continue;
+            }
+
+            String nodeName = factMapping.getFactIdentifier().getName();
+
+            DMNType rootDMNType;
+            try {
+                rootDMNType = dmnModel.getDecisionByName(nodeName) != null ?
```

Comment:
@gitgabrio 
No, the behavior is correct because user can edit the template after the creation and move a decision from EXPECT to GIVEN to test another decision and the opposite